### PR TITLE
handbook chapter about our server

### PR DESCRIPTION
**Pull request by hannesm**

### CODE REVIEW

File: src/SUMMARY.md [hunk 1]
```diff
@@ -3,3 +3,4 @@
 [Introduction](README.md)
 - [Deploy a OpenVPN server](./simple_openvpn_server.md)
 - [Deploy our MirageVPN client](./simple_miragevpn_client.md)
+- [Deploy a MirageVPN server](./miragevpn_server.md)
```
Comment (reynir) [inline src/SUMMARY.md]: We should be consistent: is it 'a' or 'our' MirageVPN x?

File: src/miragevpn_server.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
+The MirageVPN server is a unikernel which listens on a port for client
+connections, and provides Internet connectivity (with NAT) for all
+authenticated clients. A single network interface is used for both client
+connections and providing connectivity.
+
+The unikernel will encrypt all packets received from the Internet and send
+them to the respective client. All packets received from a client will be
```
Comment (reynir) [inline src/miragevpn_server.md]: Hmm. Only if there is a NAT table entry, no? I'm not sure how to phrase this better...

File: src/miragevpn_server.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+The server will route all traffic to its default gateway. If "client-to-client"
+is specified, packets for other clients will be forwarded to the specific
+client.
+
+Only a single network interface is used by the server, where both TCP listening
+and forwarding packets (using NAT) is done. NAT will always be used.
+
+## Authentication
+
+The server as is only authenticates via X.509 certificates. It is possible to
```
Comment (reynir) [inline src/miragevpn_server.md:28]: I think we don't enforce this. So presumably you could use a configuration that doesn't check certificates?! I think we "only" warn in that case.

File: src/miragevpn_server.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+["mirage"](https://mirage.io) installed on your system.
+
+## Git repository
+
+Do a `git clone https://github.com/robur-coop/miragevpn.git` to retrieve the
+source code of the MirageVPN server.
+
+## Building
+
+Inside of the cloned repository, execute `mirage configure` (other targets are
```
Comment (reynir) [inline src/miragevpn_server.md:51]: Add a reference to mirage.io on how to install mirage maybe? Or we preface with a prerequisite to have mirage installed

File: src/miragevpn_server.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+them to the respective client. All packets received from a client will be
+decrypted and forwarded to the Internet. If "client-to-client" is enabled
+in the configuration, packets from one client which destination is another
+client will be forwarded by the server.
+
+# Scope of the server
+
+The scope is at the moment limited to IPv4 (no IPv6), the server only listens
+on TCP (no UDP). Only layer 3 networking is supported (tun device), there's no
+support for tap devices.
```
Comment (reynir) [inline src/miragevpn_server.md]: The *tunnel* traffic is IPv4 only. We could listen on an IPv6 address. The server would need a IPv4 address to NAT from I guess. I'm not sure this nuance is too interesting...

File: src/miragevpn_server.md (hunk 3 above)
Comment (hannesm) [reply src/miragevpn_server.md:28]: see https://github.com/robur-coop/miragevpn/pull/265

File: src/miragevpn_server.md (hunk 4 above)
Comment (hannesm) [reply src/miragevpn_server.md:51]: There's a "prerequisite" which says "opam and mirage" need to be present -- with links to opam.ocaml.org and mirage.io. Is that sufficient?

File: src/miragevpn_server.md (hunk 5 above)
Comment (hannesm) [reply src/miragevpn_server.md]: done in 51a9dee

File: src/miragevpn_server.md (hunk 2 above)
Comment (hannesm) [reply src/miragevpn_server.md]: done in 51a9dee

File: src/SUMMARY.md (hunk 1 above)
Comment (hannesm) [reply src/SUMMARY.md]: done in 51a9dee

File: src/miragevpn_server.md (hunk 4 above)
Comment (reynir) [reply src/miragevpn_server.md:51]: Ah sorry yes that is fine and sufficient!

**Review by reynir — APPROVED** [review]

**Pull request merged**